Makefile

```diff
@@ -456,7 +456,7 @@ install-build-tools:
 
 .PHONY: install-tools
 install-tools: install-test-tools install-build-tools
-	$(GO) install github.com/vektra/mockery/v2@v2.14.0
+	$(GO) install github.com/vektra/mockery/v2@v2.42.3
```

Comment:
any reason we need to make this change? It forces too many mocks to be regenerated, not necessary for this PR.

Reply:
The previous version(`v2.14.0`) was throwing this error on running `make generate-mocks`. that's why i upgraded it.
```bash
30 Apr 24 16:30 IST INF Starting mockery dry-run=false version=v2.14.0
30 Apr 24 16:30 IST INF Walking dry-run=false version=v2.14.0
2024/04/30 16:30:08 internal error: package "encoding/base64" without types was imported from "github.com/jaegertracing/jaeger/pkg/es/client"
make: *** [Makefile:474: generate-mocks] Error 1
```

Reply:
ok, I don't mind upgrading but let's do it in a different PR and rerun for all mocks in the code.

Reply:
ok